### PR TITLE
Feature/citizen proposal list

### PR DESCRIPTION
**Pull request by martinyde**
https://jira.itkdev.dk/browse/FORSLAG-32
- Remove bold headers globally.
- Change display of citizen proposal teasers
- Add progress bar for citizen proposals.
- Fix proposals global view sorting.

### CODE REVIEW

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Plugin/Block/CitizenProposalSupportCounter.php [hunk 1]
```diff
@@ -56,16 +59,15 @@ public function getCacheMaxAge() {
...
   public function build() {
     $node = $this->routeMatch->getParameter('node');
-
-    $supportCount = $this->connection->select('hoeringsportal_citizen_proposal_support')
-      ->condition('node_id', $node->nid->value ?? 0)
-      ->countQuery()
-      ->execute()
-      ->fetchField();
+    $supportCount = $this->helper->getProposalSupportCount($node->nid->value ?? 0);
+    $supportPercentage = $supportCount ? $this->helper->calculateSupportPercentage((int) $supportCount) . '%' : '0%';
```
Comment (rimi-itk) [inline web/modules/custom/hoeringsportal_citizen_proposal/src/Plugin/Block/CitizenProposalSupportCounter.php:63]: See comments on the same code in the `Helper` class.

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php [hunk 2]
```diff
@@ -107,12 +108,26 @@ public function preprocessForm(&$variables): void {
...
+   */
+  public function preprocessNode(&$variables): void {
+    if ('citizen_proposal' !== $variables['node']->bundle()) {
+      return;
+    }
+
+    $proposalSupportCount = $this->getProposalSupportCount((int) $variables['node']->nid->value);
+
+    $variables['proposal_support_count'] = $proposalSupportCount;
+    $variables['proposal_support_percentage'] = $proposalSupportCount ? $this->calculateSupportPercentage((int) $proposalSupportCount) . '%' : '0%';
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php:122]: After reviewing the templates, I think we should pass the unformatted percentage to the templates and let the templates handle any formatting.

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php [hunk 3]
```diff
@@ -238,6 +253,42 @@ public function nodeEntityPresave(EntityInterface $entity): void {
...
+  /**
+   * Get citizen proposal support count.
+   *
+   * @param int $nid
+   *   The id of the citizen proposal to get support count for.
+   *
+   * @return mixed
+   *   A single field from the next record, or FALSE if there is no next record.
+   */
+  public function getProposalSupportCount(int $nid): mixed {
```
[the remark this answers is not in the document]
Comment (martinyde) [reply web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php:265]: To what purpose?
Future development?

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php [hunk 4]
```diff
@@ -238,6 +253,42 @@ public function nodeEntityPresave(EntityInterface $entity): void {
...
   }
 
+  /**
+   * Get citizen proposal support count.
+   *
+   * @param int $nid
+   *   The id of the citizen proposal to get support count for.
+   *
+   * @return mixed
+   *   A single field from the next record, or FALSE if there is no next record.
```
[the remark this answers is not in the document]
Comment (martinyde) [reply web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php:263]: Changed.

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Plugin/Block/CitizenProposalSupportCounter.php (hunk 1 above)
Comment (martinyde) [reply web/modules/custom/hoeringsportal_citizen_proposal/src/Plugin/Block/CitizenProposalSupportCounter.php:63]: Fixed

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php [hunk 5]
```diff
@@ -238,6 +253,42 @@ public function nodeEntityPresave(EntityInterface $entity): void {
...
+  /**
+   * Calculate the support percentage of a proposal, from the support count.
+   *
+   * @param int $proposalSupportCount
+   *   The support count af a proposal.
+   *
+   * @return int
+   *   A percentage value as an integer.
+   */
+  public function calculateSupportPercentage(int $proposalSupportCount): int {
```
[the remark this answers is not in the document]
Comment (martinyde) [reply web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php:282]: Fixed

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (martinyde) [reply web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php:122]: Changed some...

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php [hunk 6]
```diff
@@ -107,12 +108,26 @@ public function preprocessForm(&$variables): void {
...
 
+  /**
+   * Preprocess citizen proposal nodes.
+   */
+  public function preprocessNode(&$variables): void {
+    if ('citizen_proposal' !== $variables['node']->bundle()) {
+      return;
+    }
+
+    $proposalSupportCount = $this->getProposalSupportCount((int) $variables['node']->nid->value);
```
[the remark this answers is not in the document]
Comment (martinyde) [reply web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php:119]: Changed

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**